### PR TITLE
Add PER and PER Events to Diagnostics

### PR DESCRIPTION
**Pull request by martyn-w**
### Jira link

P4-2628

### What?

I have added/removed/altered:

- [x] Added PER and PER events to move diagnostics
- [x] Tweaked wording of events "timestamp -> occurred_at"

### Why?

I am doing this because:

- so suppliers can see PER events they are raising for testing and development

### Have you? (optional)

- [ ] Updated API docs if necessary	
- [ ] Added environment variables to [deployment repository](https://github.com/ministryofjustice/hmpps-book-secure-move-api-deploy)

### Deployment risks (optional)

- minimal

### CODE REVIEW

File: app/services/diagnostics/move_inspector.rb [hunk 1]
```diff
@@ -37,13 +37,13 @@ def generate
       ENDMOVEEVENTS
 
       if move.generic_events.any?
-        @output << "#{'EVENT'.ljust(30)}\t#{'TIMESTAMP'.ljust(27)}\t#{'NOTES'.ljust(30)}\tDETAILS\n"
+        @output << "#{'EVENT'.ljust(30)}\t#{'OCCURRED AT'.ljust(27)}\t#{'NOTES'.ljust(30)}\tDETAILS\n"
         move.generic_events.applied_order.each do |event| # NB use each to preserve sort order
           # NB only show event params if include_person_details==true, as they could contain personal details
-          @output << "#{event.type.ljust(30)}\t#{event.occurred_at.to_s.ljust(27)}\t#{include_person_details ? event.notes.to_s.truncate(30).ljust(30) : '-'.ljust(30)}\t#{include_person_details ? event.details : '-'}\n"
+          @output << "#{event.type.delete_prefix('GenericEvent::').ljust(30)}\t#{event.occurred_at.to_s.ljust(27)}\t#{include_person_details ? event.notes.to_s.truncate(30).ljust(30) : '-'.ljust(30)}\t#{include_person_details ? event.details : '-'}\n"
```
Comment (cwrw) [inline app/services/diagnostics/move_inspector.rb:43]: I think you can use `event_type` here instead, which removes the prefix: https://github.com/ministryofjustice/hmpps-book-secure-move-api/blob/master/app/models/generic_event.rb#L116

File: app/services/diagnostics/move_inspector.rb [hunk 2]
```diff
@@ -61,19 +61,19 @@ def generate
...
       if move.journeys.any?
         move.journeys.default_order.each do |journey| # NB use each to preserve sort order
           @output << "#{journey.from_location.title} --> #{journey.to_location.title} (#{journey.id})\n"
           if journey.generic_events.any?
-            @output << "  #{'EVENT'.ljust(30)}\t#{'TIMESTAMP'.ljust(27)}\t#{'NOTES'.ljust(30)}\tDETAILS\n"
+            @output << "  #{'EVENT'.ljust(30)}\t#{'OCCURRED AT'.ljust(27)}\t#{'NOTES'.ljust(30)}\tDETAILS\n"
             journey.generic_events.applied_order.each do |event| # NB use each to preserve sort order
               # NB only show event params if include_person_details==true, as they could contain personal details
-              @output << "  #{event.type.ljust(30)}\t#{event.occurred_at.to_s.ljust(27)}\t#{include_person_details ? event.notes.to_s.truncate(30).ljust(30) : '-'.ljust(30)}\t#{include_person_details ? event.details : '-'}\n"
+              @output << "  #{event.type.delete_prefix('GenericEvent::').ljust(30)}\t#{event.occurred_at.to_s.ljust(27)}\t#{include_person_details ? event.notes.to_s.truncate(30).ljust(30) : '-'.ljust(30)}\t#{include_person_details ? event.details : '-'}\n"
```
Comment (cwrw) [inline app/services/diagnostics/move_inspector.rb:76]: same here re `event_type`

File: app/services/diagnostics/move_inspector.rb (hunk 1 above)
Comment (martyn-w) [reply app/services/diagnostics/move_inspector.rb:43]: thanks, didn't know about that 👍

File: app/services/diagnostics/move_inspector.rb [hunk 3]
```diff
@@ -171,6 +171,50 @@ def generate
...
+          PERSON ESCORT RECORD
+          --------
+        ENDPER
+
+        # NB: it is better to identify PERs via profile (not via move directly), as there are some older records which are
+        # only associated with a profile but not with a move
+        if move.profile&.person_escort_record.present?
+          move.profile.person_escort_record.tap do |per|
+            @output << "id:\t#{per.id}\n"
+            @output << "framework_id:\t#{per.framework_id}\n"
```
Comment (cwrw) [inline app/services/diagnostics/move_inspector.rb:186]: might be good to also show the framework version as well as id

File: app/services/diagnostics/move_inspector.rb [hunk 4]
```diff
@@ -171,6 +171,51 @@ def generate
...
+          PERSON ESCORT RECORD
+          --------
+        ENDPER
+
+        # NB: it is better to identify PERs via profile (not via move directly), as there are some older records which are
+        # only associated with a profile but not with a move
+        if move.profile&.person_escort_record.present?
+          move.profile.person_escort_record.tap do |per|
+            @output << "id:\t#{per.id}\n"
+            @output << "framework ver:\t#{per.framework&.version}\n"
```
Comment (cwrw) [inline app/services/diagnostics/move_inspector.rb:186]: ✂️  ver/versions :)

**Review by cwrw — APPROVED** [review]
just the ver/versions typo otherwise looks good

File: app/services/diagnostics/move_inspector.rb (hunk 4 above)
Comment (martyn-w) [reply app/services/diagnostics/move_inspector.rb:186]: "version" was a bit too long vs "ver", and it made the tab indenting a little ugly:

```
PERSON ESCORT RECORD
--------
id:	14f3744d-edf3-4ab0-af34-107bbebc0ed0
framework version:	1.3.0
framework_id:	098cec4e-e0b5-47c2-988d-16efbf600a4e
profile_id:	2bbb14e5-0868-4d5a-8dad-7ba825aea468
move_id:	f24e4449-61b7-425c-84a2-60be57dca5a1
```

vs

```
PERSON ESCORT RECORD
--------
id:	14f3744d-edf3-4ab0-af34-107bbebc0ed0
framework ver:	1.3.0
framework_id:	098cec4e-e0b5-47c2-988d-16efbf600a4e
profile_id:	2bbb14e5-0868-4d5a-8dad-7ba825aea468
move_id:	f24e4449-61b7-425c-84a2-60be57dca5a1
```

Comment (martyn-w) [reply app/services/diagnostics/move_inspector.rb:186]: but I'm not too fussy either way :-)

Comment (cwrw) [reply app/services/diagnostics/move_inspector.rb]: ah ok fair enough, merge away

**Pull request merged**